lean/commands/live/deploy.py

```diff
@@ -369,7 +374,12 @@ def deploy(project: Path,
     project_config = project_config_manager.get_project_config(algorithm_file.parent)
     engine_image = cli_config_manager.get_engine_image(image or project_config.get("engine-image", None))
 
-    container.update_manager.pull_docker_image_if_necessary(engine_image, update)
+    if no_update and update:
+        logger.warn("Both --no-update and --update are specified, ignoring --no-update")
+        container.update_manager.pull_docker_image_if_necessary(engine_image, update)
+
+    if not no_update:
+        container.update_manager.pull_docker_image_if_necessary(engine_image, update)
```

Comment:
Given we pass `update` we could pass in `no_update` too and handle it all internally, would avoid the duplication?